### PR TITLE
CY-874 default import resolver: fallback - make configurable

### PR DESCRIPTION
**Pull request by tehasdf**
Falling back to the original url can now be configured from the
pctx level

### CODE REVIEW

File: dsl_parser/import_resolver/default_import_resolver.py [hunk 1]
```diff
@@ -70,23 +70,29 @@ class DefaultImportResolver(AbstractImportResolver):
 
         In case that all the resolve attempts will fail,
         a DSLParsingLogicException will be raise.
+
+    If ``fallback`` is set (the default), then in case all rules fail to
+    be resolved, the original url will be tried anyway.
     """
 
-    def __init__(self, rules=None):
+    def __init__(self, rules=None, fallback=True):
```
Comment (isaac-s) [inline dsl_parser/import_resolver/default_import_resolver.py]: `False` should be the default as it's more intuitive.

Comment (isaac-s) [reply dsl_parser/import_resolver/default_import_resolver.py]: This makes the resolver configurable at the code level, but not in a way that is usable for the customer. There should be another PR to `cloudify-manager-install` with the appropriate change in `config.yaml`.

Comment (tehasdf) [reply dsl_parser/import_resolver/default_import_resolver.py]: @isaac-s note that the restservice creates the import resolver by just doing `resolver(**kw)` with args coming from the provider context; so this is still half-accessible by the customer, by changing the provider context

Comment (isaac-s) [reply dsl_parser/import_resolver/default_import_resolver.py]: @tehasdf customers shouldn't, and won't, modify the provider context.
Let alone the fact that we'd like to get rid of the pctx altogether.

**Comment by tehasdf** [comment]
Added a `config.yaml` frontend for this in cloudify-cosmo/cloudify-manager-install#401

File: dsl_parser/import_resolver/default_import_resolver.py (hunk 1 above)
Comment (geokala) [reply dsl_parser/import_resolver/default_import_resolver.py]: What's the current behaviour (and is the current behaviour something that we should consider erroneous)?

Comment (tehasdf) [reply dsl_parser/import_resolver/default_import_resolver.py]: current behaviour is like True, ie. if the resolved url returns a 404, still try the original url. We decided that's wrong and it should just be an error instead (ie. like False)

**Pull request merged**